Fix the build of libetonyek / libreoffice:

https://issues.guix.gnu.org/72040

Patch copied from upstream source repository:

https://github.com/boostorg/phoenix/commit/8913607a3788cb82d48ed461ea59c919b7bad3df

From 8913607a3788cb82d48ed461ea59c919b7bad3df Mon Sep 17 00:00:00 2001
From: djowel <djowel@gmail.com>
Date: Tue, 29 Aug 2023 14:32:41 +0800
Subject: [PATCH] Disabled this for now, due to ODR violations $$$ Fix Me $$$

---
 include/boost/phoenix/stl/tuple.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/boost/phoenix/stl/tuple.hpp b/include/boost/phoenix/stl/tuple.hpp
index fb9440d2..a2e42bec 100644
--- a/include/boost/phoenix/stl/tuple.hpp
+++ b/include/boost/phoenix/stl/tuple.hpp
@@ -106,6 +106,7 @@ namespace boost { namespace phoenix {
         tuple_detail::idx_wrap<N>(), t);
     }
 
+#if 0 // Disabled this for now due to ODR viaolations $$$ Fix Me $$$
     // Make unpacked argument placeholders
     namespace placeholders {
         #define BOOST_PP_LOCAL_LIMITS (1, BOOST_PHOENIX_ARG_LIMIT)
@@ -114,6 +115,8 @@ namespace boost { namespace phoenix {
             boost::phoenix::get_<(N)-1>(boost::phoenix::placeholders::arg1);
         #include BOOST_PP_LOCAL_ITERATE()
     }
+#endif
+
 }} // namespace boost::phoenix
 
 #endif // C++ 14
